### PR TITLE
Add output-path security evaluation and group-aware permission checks

### PR DESCRIPTION
**Pull request by isseis**
This pull request implements comprehensive path validation functionality for output files, adds robust test coverage, and improves security configuration handling. The most important changes are the addition of a new `DefaultPathValidator` for path validation and traversal prevention, extensive unit tests for path validation, and updates to security configuration usage in runner tests. Additionally, a minor improvement was made to directory permission validation logic.

### Output Path Validation

* Added `DefaultPathValidator` in `internal/runner/output/path.go` for basic validation and normalization of output paths, including absolute/relative path checks, traversal prevention, and work directory containment.
* Marked all path processing implementation and test tasks as completed in `docs/tasks/0025_command_output/04_implementation_plan.md`, reflecting the new validator and its coverage.

### Testing

* Added comprehensive unit tests for `DefaultPathValidator` in `internal/runner/output/path_test.go`, covering valid paths, traversal attempts, work directory handling, and edge cases.

### Security Configuration Handling

* Updated runner tests in `internal/runner/runner_test.go` to use `security.DefaultConfig()` and override only necessary fields, improving maintainability and consistency. [[1]](diffhunk://#diff-3b8985514864e913f3a3b3463fc897dd19897655c1b71ae44b0928b5b5a94f7fL140-R143) [[2]](diffhunk://#diff-3b8985514864e913f3a3b3463fc897dd19897655c1b71ae44b0928b5b5a94f7fL155-R156) [[3]](diffhunk://#diff-3b8985514864e913f3a3b3463fc897dd19897655c1b71ae44b0928b5b5a94f7fL170-R169) [[4]](diffhunk://#diff-3b8985514864e913f3a3b3463fc897dd19897655c1b71ae44b0928b5b5a94f7fL194-R191) [[5]](diffhunk://#diff-3b8985514864e913f3a3b3463fc897dd19897655c1b71ae44b0928b5b5a94f7fL211-R206)

### Directory Permission Validation

* Improved directory permission check in `internal/runner/security/file_validation.go` to allow bypassing world-writable checks only if explicitly configured for permissive testing.
* Added missing imports in `internal/runner/security/file_validation.go` for user and string handling.

### CODE REVIEW

File: internal/runner/output/path.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+		}
+
+		// Check for other control characters not in lowRiskChars
+		if unicode.IsControl(r) && !slices.Contains(lowRiskChars, runeStr) {
+			addDangerous(runeStr)
+		}
+	}
+
+	return found
+}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/output/path.go:196]: revised

File: internal/runner/output/path.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+// ValidateAndResolvePath validates and resolves an output path
+// This performs basic path validation and path traversal prevention
+// Additional security checks (symlink detection, etc.) are performed by SecurityValidator
+func (v *DefaultPathValidator) ValidateAndResolvePath(outputPath, workDir string) (string, error) {
+	// Trim whitespace and check for empty path
+	if outputPath == "" {
+		return "", ErrEmptyPath
+	}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/output/path.go:38]: path name containing space characters should fail in the validation. We'll call ValidateAndResolvePath() before actually open the output file in a coming phase.
